tests/post/steps/test_monitoring.py

```diff
@@ -179,4 +192,63 @@ def _node_has_metrics():
     utils.retry(_node_has_metrics, times=60, wait=3)
 
 
+@then(
+    "the deployed Prometheus alert rules are the same as the default "
+    "alert rules")
+def check_deployed_rules(host, prometheus_api):
+    def _wait_prom_rule_cmp():
+        try:
+            deployed_rules = prometheus_api.get_rules()
+        except utils.PrometheusApiError as exc:
+            pytest.fail(str(exc))
+
+        if not deployed_rules:
+            pytest.fail(
+                "Expected default deployed Prometheus rules but got {}"
+                .format(deployed_rules)
```

Comment:
Don't see the point of showing the content of `deployed_rules`, with the `if` before, we know its empty.
Moreover, I'm not even sure that this case can happen since it's the return from the API, so you should at least have the status and the `data` key.
For me, you can remove this whole `if` and let the assertion fails at the bottom of this function.